### PR TITLE
Fix some bugs and cleaning code

### PR DESCRIPTION
**Pull request by WhyZerVellasskx**

### CODE REVIEW

File: src/main/java/ru/abstractmenus/AbstractMenus.java [hunk 1]
```diff
@@ -57,29 +58,24 @@
...
 import java.util.Optional;
 
 @Getter
 public final class AbstractMenus extends JavaPlugin implements AbstractMenusPlugin {
 
     private static AbstractMenus instance;
 
+    @Getter
     private CommandManager commandManager;
+    @Getter
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/AbstractMenus.java]: Why?
You already have @Getter above class

File: src/main/java/ru/abstractmenus/AbstractMenus.java [hunk 2]
```diff
@@ -229,9 +226,9 @@ private void registerCommands() {
                 .addSub("mul", new VarpMul())
                 .addSub("div", new VarpDiv());
 
-        getServer().getPluginCommand("am").setExecutor(am);
-        getServer().getPluginCommand("var").setExecutor(var);
-        getServer().getPluginCommand("varp").setExecutor(varp);
+        Objects.requireNonNull(getServer().getPluginCommand("am")).setExecutor(am);
+        Objects.requireNonNull(getServer().getPluginCommand("var")).setExecutor(var);
+        Objects.requireNonNull(getServer().getPluginCommand("varp")).setExecutor(varp);
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/AbstractMenus.java:229]: Why?

File: src/main/java/ru/abstractmenus/data/actions/ActionPlacedItemRemove.java [hunk 3]
```diff
@@ -14,8 +14,8 @@
 /**
  * removePlaced: [slot]
  * removePlaced: {
- *     slot: [slot]
- *     count: [int]
+ * slot: [slot]
+ * count: [int]
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/actions/ActionPlacedItemRemove.java]: Need tab/whitespace here. Rollback please

File: src/main/java/ru/abstractmenus/data/activators/OpenButton.java [hunk 4]
```diff
@@ -25,11 +26,11 @@ private OpenButton(List<TypeLocation> location) {
     public void onButtonClick(PlayerInteractEvent event) {
         if (!ActivatorUtil.checkHand(event)) return;
 
-        if(event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
-                && event.getClickedBlock().getType().toString().contains("BUTTON")){
+        if (event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                && Objects.requireNonNull(event.getClickedBlock()).getType().toString().contains("BUTTON")) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenButton.java]: Maybe `event.getClickedBlock != null`?

File: src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java [hunk 5]
```diff
@@ -18,23 +18,23 @@ public class OpenClickItem extends Activator {
...
+    private OpenClickItem(List<Item> items) {
         this.items = items;
     }
 
     @EventHandler
-    public void onInteract(PlayerInteractEvent event){
-        if(items != null && event.getItem() != null){
-            if(event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK){
-                if(ActivatorUtil.checkHand(event)) {
+    public void onInteract(PlayerInteractEvent event) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java:29]: Need rework :D

File: src/main/java/ru/abstractmenus/data/activators/OpenCommand.java [hunk 6]
```diff
@@ -67,23 +65,33 @@ private void onCommand(PlayerCommandPreprocessEvent event) {
...
+//            this.activator = activator;
+//        }
+//
+//        @Override
+//        public void handle(CommandSender sender, CommandContext ctx) {
+//            if (sender instanceof Player) {
+//                activator.openMenu(ctx, (Player) sender);
+//            }
+//        }
+//    }
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenCommand.java]: Remove this

File: src/main/java/ru/abstractmenus/data/activators/OpenCommand.java [hunk 7]
```diff
@@ -67,23 +65,33 @@ private void onCommand(PlayerCommandPreprocessEvent event) {
...
-        }
+    private record Handler(OpenCommand activator) implements CommandHandler {
 
         @Override
-        public void handle(CommandSender sender, CommandContext ctx) {
-            if (sender instanceof Player) {
-                activator.openMenu(ctx, (Player) sender);
+            public void handle(CommandSender sender, CommandContext ctx) {
+                if (sender instanceof Player) {
+                    activator.openMenu(ctx, (Player) sender);
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenCommand.java]:
```suggestion
                if (sender instanceof Player player) {
                    activator.openMenu(ctx, player);
```

File: src/main/java/ru/abstractmenus/data/activators/OpenLever.java [hunk 8]
```diff
@@ -13,21 +13,22 @@
...
     }
 
     @EventHandler
     public void onLeverClick(PlayerInteractEvent event) {
         if (!ActivatorUtil.checkHand(event)) return;
 
-        if(event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
-                && event.getClickedBlock().getType().equals(Material.LEVER)) {
+        if (event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                && Objects.requireNonNull(event.getClickedBlock()).getType().equals(Material.LEVER)) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenLever.java]: same

File: src/main/java/ru/abstractmenus/data/activators/OpenPlate.java [hunk 9]
```diff
@@ -12,23 +12,24 @@
...
     }
 
     @EventHandler
     public void onPlateEnter(PlayerInteractEvent event) {
         if (!ActivatorUtil.checkHand(event)) return;
 
-        if(event.getAction().equals(Action.PHYSICAL)){
-            if(event.getClickedBlock().getType().toString().contains("PLATE")){
+        if (event.getAction().equals(Action.PHYSICAL)) {
+            if (Objects.requireNonNull(event.getClickedBlock()).getType().toString().contains("PLATE")) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenPlate.java]: same

File: src/main/java/ru/abstractmenus/events/RegionEvent.java [hunk 10]
```diff
@@ -1,30 +1,26 @@
...
 
-    public ProtectedRegion getRegion(){
-        return region;
-    }
-
-    public Player getPlayer(){
-        return player;
-    }
-
+    @NotNull
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/events/RegionEvent.java:23]: Seems like a useless annotation

File: src/main/java/ru/abstractmenus/extractors/EntityExtractor.java
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/extractors/EntityExtractor.java:1]: If you change this - change correctly to:
```java

        if (obj instanceof Entity entity) {
            if (entity instanceof LivingEntity livingEntity) {
                return switch (placeholder) {
                    case "entity_last_damage" -> String.valueOf(livingEntity.getLastDamage());
                    case "entity_no_damage_ticks" -> String.valueOf(livingEntity.getNoDamageTicks());
                    case "entity_killer" ->
                            livingEntity.getKiller() != null ? livingEntity.getKiller().getName() : null;
                    case "entity_eye_height" -> String.valueOf(livingEntity.getEyeHeight());
                    default -> "";
                };
            }

            return switch (placeholder) {
                case "entity_type" -> entity.getType().toString();
                case "entity_id" -> String.valueOf(entity.getEntityId());
                case "entity_uuid" -> entity.getUniqueId().toString();
                case "entity_name" -> entity.getName();
                case "entity_custom_name" -> entity.getCustomName();
                case "entity_world" -> entity.getWorld().getName();
                case "entity_loc_x" -> String.valueOf(entity.getLocation().getX());
                case "entity_loc_y" -> String.valueOf(entity.getLocation().getY());
                case "entity_loc_z" -> String.valueOf(entity.getLocation().getZ());
                case "entity_pose" -> entity.getPose().toString(); // 1.14+
                case "entity_facing" -> entity.getFacing().toString();
                case "entity_ticks_lived" -> String.valueOf(entity.getTicksLived());
                default -> "";
            };
        }
```

File: src/main/java/ru/abstractmenus/extractors/ItemStackExtractor.java [hunk 11]
```diff
@@ -15,20 +15,28 @@ public String extract(Object obj, String placeholder) {
             ItemStack item = (ItemStack) obj;
 
             switch (placeholder) {
-                case "item_type": return item.getType().toString();
-                case "item_data": return String.valueOf(item.getData().getData()); // 1.12-
-                case "item_amount": return String.valueOf(item.getAmount());
-                case "item_max_stack": return String.valueOf(item.getMaxStackSize());
-                case "item_serialized": return ItemUtil.encodeStack(item);
+                case "item_type":
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/extractors/ItemStackExtractor.java]: Same. Need change to switch expression

File: src/main/java/ru/abstractmenus/listeners/InventoryListener.java [hunk 12]
```diff
@@ -142,7 +142,7 @@ public void onInventoryClose(InventoryCloseEvent event) {
             Menu current = MenuManager.instance().getOpenedMenu(player);
 
             if (closed instanceof AbstractMenu) {
-                ((AbstractMenu)closed).dropPlaced(player);
+                ((AbstractMenu) closed).dropPlaced(player);
             }
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/listeners/InventoryListener.java]:
```java
 if (closed instanceof AbstractMenu abstractMenu) {
     abstractMenu.dropPlaced(player);
 }
```

File: src/main/java/ru/abstractmenus/menu/SimpleMenu.java [hunk 13]
```diff
@@ -78,7 +78,7 @@ protected void placeItems(Player player) {
                     if (built.getAmount() > 0) {
                         Slot slot = ((InventoryItem) item).getSlot(player, this);
 
-                        if (slot instanceof SlotIndex && ((SlotIndex)slot).getIndex() == -1) {
+                        if (slot instanceof SlotIndex && ((SlotIndex) slot).getIndex() == -1) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/menu/SimpleMenu.java:81]:
```suggestion
                        if (slot instanceof SlotIndex slotIndex && slotIndex.getIndex() == -1) {
```

Comment (BrainRTP) [reply src/main/java/ru/abstractmenus/menu/SimpleMenu.java:81]: There are plenty of moments like this in this class. 
There are also a lot of them by code.

File: src/main/java/ru/abstractmenus/placeholders/hooks/CatalogPlaceholders.java [hunk 14]
```diff
@@ -17,11 +17,16 @@ public String replace(String placeholder, Player player) {
...
+                    return String.valueOf(tmpl.getPage());
+                case "pages":
+                    return String.valueOf(tmpl.getPages());
+                case "page_next":
+                    return String.valueOf(tmpl.getPage() + 1);
+                case "page_prev":
+                    return String.valueOf(tmpl.getPage() - 1);
+                case "elements":
+                    return String.valueOf(tmpl.getSnapshot().size());
             }
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/placeholders/hooks/CatalogPlaceholders.java]: Change to switch expression

**Review by BrainRTP — CHANGES_REQUESTED** [review]

File: src/main/java/ru/abstractmenus/AbstractMenus.java (hunk 2 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/AbstractMenus.java:229]: Intellij idea asked me to do this. xd In general, the command cannot be empty, so it is almost a creeping feature, but there are fewer varnishes ;D

File: src/main/java/ru/abstractmenus/data/activators/OpenButton.java (hunk 4 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/data/activators/OpenButton.java]: okay

File: src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java (hunk 5 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java:29]: ;D

File: src/main/java/ru/abstractmenus/AbstractMenus.java (hunk 1 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/AbstractMenus.java]: And I really will fix it now

File: src/main/java/ru/abstractmenus/listeners/InventoryListener.java (hunk 12 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/listeners/InventoryListener.java]: okay

File: src/main/java/ru/abstractmenus/extractors/ItemStackExtractor.java (hunk 11 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/extractors/ItemStackExtractor.java]: I didn't really understand what you meant + I didn't touch the logic of the code, I just used CTRL + ALT + L to format the code automatically

File: src/main/java/ru/abstractmenus/menu/SimpleMenu.java (hunk 13 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/menu/SimpleMenu.java:81]: ????

File: src/main/java/ru/abstractmenus/extractors/ItemStackExtractor.java (hunk 11 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/extractors/ItemStackExtractor.java]: And I understood

File: src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java (hunk 5 above)
Comment (WhyZerVellasskx) [reply src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java:29]: okay

Comment (BrainRTP) [reply src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java:29]: my bad, it's like //todo 
Not a call to action 😨

File: src/main/java/ru/abstractmenus/menu/SimpleMenu.java (hunk 13 above)
Comment (BrainRTP) [reply src/main/java/ru/abstractmenus/menu/SimpleMenu.java:81]: Forget it, I got a little carried away. It doesn't require edits as part of this PR

File: src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java (hunk 5 above)
Comment (BrainRTP) [reply src/main/java/ru/abstractmenus/data/activators/OpenClickItem.java:29]: But thank you

File: src/main/java/ru/abstractmenus/data/activators/OpenLever.java [hunk 15]
```diff
@@ -13,21 +13,23 @@
...
 
     @EventHandler
     public void onLeverClick(PlayerInteractEvent event) {
         if (!ActivatorUtil.checkHand(event)) return;
+        if (event.getClickedBlock() == null) return;
 
-        if(event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
-                && event.getClickedBlock().getType().equals(Material.LEVER)) {
+        if (event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                && (event.getClickedBlock()).getType().equals(Material.LEVER)) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenLever.java]:
```suggestion
                && event.getClickedBlock().getType().equals(Material.LEVER)) {
```

File: src/main/java/ru/abstractmenus/data/activators/OpenPlate.java [hunk 16]
```diff
@@ -12,23 +12,25 @@
...
 
     @EventHandler
     public void onPlateEnter(PlayerInteractEvent event) {
         if (!ActivatorUtil.checkHand(event)) return;
+        if (event.getClickedBlock() == null) return;
 
-        if(event.getAction().equals(Action.PHYSICAL)){
-            if(event.getClickedBlock().getType().toString().contains("PLATE")){
+        if (event.getAction().equals(Action.PHYSICAL)) {
+            if ((event.getClickedBlock()).getType().toString().contains("PLATE")) {
```
Comment (BrainRTP) [inline src/main/java/ru/abstractmenus/data/activators/OpenPlate.java]:
```suggestion
            if (event.getClickedBlock().getType().toString().contains("PLATE")) {
```

**Review by BrainRTP — APPROVED** [review]

**Pull request merged**